pkg/cloud/azure/services/availabilitysets/availabilitysets.go

```diff
@@ -92,3 +98,35 @@ func (s *Service) Delete(ctx context.Context, spec azure.Spec) error {
 
 	return nil
 }
+
+// getMaximumFaultDomainCount retrieves the MaximumPlatformFaultDomainCount from the SKU service.
+func (s *Service) getMaximumFaultDomainCount(ctx context.Context) (int, error) {
+	skuService := resourceskus.NewService(s.Scope)
+	skuSpec := resourceskus.Spec{
+		Name:         string(compute.AvailabilitySetSkuTypesAligned),
+		ResourceType: resourceskus.AvailabilitySets,
+	}
+
+	defaultFaultDomainCount := 2
+	skuI, err := skuService.Get(ctx, skuSpec)
+	if err != nil {
+		if errors.Is(err, resourceskus.ErrResourceNotFound) {
+			return defaultFaultDomainCount, fmt.Errorf("failed to find aligned SKU '%s' in Azure: %w", skuSpec.Name, err)
+		}
+		return defaultFaultDomainCount, fmt.Errorf("failed to retrieve SKU information: %w", err)
```

Comment:
It would be helpful to also have the `skuSpec.Name` in this error, as well.

Reply:
Thank you @nrb updated.